lib/utils/search.js

```diff
@@ -32,12 +38,47 @@ const getWeight = (searchTerm, words, resource, resourceIndex) => {
   return weight;
 };
 
+const weightByCategories = (ids, items) => {
+  return JSON.parse(JSON.stringify(items)).map(item => {
+    item.resources = item.resources.map(resource => {
+      resource.weight = getCategoryWeight(ids, items, resource);
+      return resource;
+    });
+    return item;
+  });
+};
+
+const getCategoryWeight = (ids, categorisedResources, resource) => {
+  const categoriesResourceIsIn = categorisedResources
+    .filter(category => {
+      return category.resources.some(r => r.id == resource.id);
+    })
+    .map(x => x.id);
+
+  const selectedCategoriesResourceIsIn = ids.filter(id => {
+    return categoriesResourceIsIn.includes(id);
+  });
```

Comment:
I think you can do just: `const selectedCategoriesResourceIsIn = ids.filter(id => categoriesResourceIsIn.includes(id));`

Reply:
Same for the statement Ln62-64 bellow.